config/routes.rb

```diff
@@ -86,13 +86,16 @@
 
   resource :core_induction_programme, controller: :core_induction_programme, only: :show, as: "cip", path: "/core-induction-programme" do
     get "download-export", to: "core_induction_programme#download_export", as: "download_export"
-    resource :years, controller: "core_induction_programme/years", only: :show, path: "/:year_id" do
-      resource :modules, controller: "core_induction_programme/modules", only: :show, path: "/:module_id" do
-        resource :lessons, controller: "core_induction_programme/lessons", only: %i[show edit update], path: "/:lesson_id" do
-          member do
-            put "edit"
-          end
-        end
+
+    concern :editable do
+      member do
+        put "edit"
+      end
+    end
```

Comment:
Interesting, what do we need it for?

Reply:
As i understand it, it adds the put "edit" in to each nested resource, reducing the duplication of putting a member in each resource.

Reply:
Yeah, if we remove it the test ```put "#{course_lesson_url}/edit"``` fails, saying there's no path that matches. 

The editable concern adds  
```edit_cip_year_module_lesson PUT    /core-induction-programme/years/:year_id/modules/:module_id/lessons/:id/edit(.:format)            core_induction_programme/lessons#edit```. Without it we only have a get with this path. 

I had a play around with it to try and not use it but I think preview rendered again but it removed the /edit from the end of the url still in the preview. This was the way that I found keeps it in.

Reply:
The concern has been removed as it wasn't necessary to include.